File: Hive.Domain/GameState.cs
using System;
using System.Collections.Generic;
using System.Runtime.CompilerServices;

namespace Hive.Domain
{
    public class GameState
    {
        public ICollection<Cell> Cells { get; } = new List<Cell>();
        public ICollection<Player> Players { get; } = new List<Player>();
    }
}

File: src/Hive.FrontEnd/src/domain/game-state.ts
import { Cells } from './cell';
import { Players } from './player';
import { HistoricalMoves } from './historical-move';

export type GameState = {
  cells: Cells;
  players: Players;
  gameId: string;
  gameStatus: GameStatus;
  history: HistoricalMoves;
};

export type GameStatus =
  | 'NewGame'
  | 'MoveSuccess'
  | 'AiWin'
  | 'Player0Win'
  | 'Player1Win'
  | 'MoveSuccessNextPlayerSkipped'
  | 'MoveInvalid'
  | 'GameOver'
  | 'Draw';
